consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -176,149 +176,133 @@ impl<
             return false;
         }
 
-        // Update ledger_sync_state.
-        let sync_service_is_behind = self.ledger_sync_service.is_behind(&self.network_state);
-        match (self.ledger_sync_state.clone(), sync_service_is_behind) {
-            // Fully in sync, nothing to do.
-            (LedgerSyncState::InSync, false) => {}
-
-            // Sync service reports we're behind and we are just finding out about it now.
-            (LedgerSyncState::InSync, true) => {
-                log::info!(self.logger, "sync_service reported we are behind, we're at slot {} and network state is {:?}", self.current_slot_index, self.network_state.peer_to_current_slot());
-                self.ledger_sync_state = LedgerSyncState::MaybeBehind(Instant::now());
+        // Advance the "sync state" state machine.
+        let previous_sync_state = {
+            let next_state = self.next_sync_state(Instant::now());
+            std::mem::replace(&mut self.ledger_sync_state, next_state)
+        };
+        let mut should_sync = false;
+        match (previous_sync_state, &self.ledger_sync_state) {
```

Comment:
thinking ahead to [MCC-1720](https://mobilecoin.atlassian.net/browse/MCC-1720) - will we be adding an additional state (e.g. "IsAhead") to manage a delay after externalizing a block "too quickly"? Is there a better way to manage the transition table than a big match?

Reply:
I'm hoping that this can be whittled down. This PR will mostly focus on testability, though, and i don't expect it to change much functionality.